File: rust/5_kyu/rgb_to_hex_conversion.rs
/*
SPDX-FileCopyrightText: 2025 Michael Persico <michael.a.persico@gmail.com>
 
SPDX-License-Identifier: MIT-0
*/

// https://www.codewars.com/kata/513e08acc600c94f01000001

fn rgb(r: i32, g: i32, b: i32) -> String {
  format!("{:02X}{:02X}{:02X}", r.clamp(0, 255), g.clamp(0, 255), b.clamp(0, 255))
}

fn main() {
    println!("{}", rgb(0,0,0)); // 000000
    println!("{}", rgb(1,2,3)); // 010203
    println!("{}", rgb(255,255,255)); // FFFFFF
    println!("{}", rgb(254,253,252)); // FEFDFC
    println!("{}", rgb(-20,275,125)); // 00FF7D
}

File: go/5_kyu/rgb_to_hex_conversion.go
/*
SPDX-FileCopyrightText: 2025 Michael Persico <michael.a.persico@gmail.com>

SPDX-License-Identifier: MIT-0
*/

// https://www.codewars.com/kata/513e08acc600c94f01000001

packa
package main

import (
	"fmt"
)

func RGB(r, g, b int) string {
	convert := func(num int) string {
		switch {
		case num < 0:
			return "00"
		case num > 255:
			return "FF"
		default:
			return fmt.Sprintf("%02X", num)
		}
	}
	return convert(r) + convert(g) + convert(b)
}

func main() {
	fmt.Println(RGB(0, 0, 0))       // 000000
	fmt.Println(RGB(1, 2, 3))       // 010203
	fmt.Println(RGB(255, 255, 255)) // FFFFFF
	fmt.Println(RGB(254, 253, 252)) // FEFDFC
	fmt.Println(RGB(-20, 275, 125)) //00FF7D
}
